File: subprojects/frida-node/examples/bytecode.ts
import frida from "frida";

const processName = process.argv[2];

const source = `
rpc.exports = {
  listThreads() {
    return Process.enumerateThreads();
  }
};
`;

async function main() {
    const systemSession = await frida.attach(0);
    const bytecode = await systemSession.compileScript(source, {
        name: "bytecode-example"
    });

    const session = await frida.attach(processName);
    const script = await session.createScriptFromBytes(bytecode);
    script.message.connect(message => {
        console.log("[*] Message:", message);
    });
    await script.load();

    console.log("[*] Called listThreads() =>", await script.exports.listThreads());

    await script.unload();
}

main()
    .catch(e => {
        console.error(e);
    });

File: subprojects/frida-python/examples/bytecode.py
import frida

system_session = frida.attach(0)
bytecode = system_session.compile_script(
    name="bytecode-example",
    source="""\
rpc.exports = {
  listThreads: function () {
    return Process.enumerateThreadsSync();
  }
};
""",
)

session = frida.attach("Twitter")
script = session.create_script_from_bytes(bytecode)
script.load()
api = script.exports_sync
print("api.list_threads() =>", api.list_threads())
